tuf_helpers.py

```diff
@@ -0,0 +1,78 @@
+
+import os
+import pathlib
+import shutil
+from typing import Dict
+
+import securesystemslib as sslib
+import tuf.scripts.repo as tuf_repo
+import tuf.repository_tool as repo_tool
+
+
+default_trust_dir = pathlib.Path.joinpath(pathlib.Path.home(), '.pysigny')
+keystore_dir = 'private'
+root_key_name = 'root_key'
+targets_key_name = 'targets_key'
+snapshot_key_name = 'snapshot_key'
+timestamp_key_name = 'timestamp_key'
+metadata_staged_dir = 'metadata.staged'
+metadata_dir = 'metadata'
+
+
+class TUFKey:
+    private: None
+    public: None
+    passphrase: None
+
+
+keys: Dict[str, TUFKey] = {
+    root_key_name: TUFKey,
+    targets_key_name: TUFKey,
+    snapshot_key_name: TUFKey,
+    timestamp_key_name: TUFKey
+}
+
+
+def init_repo(target, name, trustdir):
+    repo_path = pathlib.Path.joinpath(trustdir, name)
+    repository = repo_tool.create_new_repository(str(repo_path))
+    create_and_set_keys(repository, name, trustdir)
+    write_repo(name, trustdir)
+
+
+def create_and_set_keys(repository, name, trustdir):
+    for k in keys:
+        p = os.environ.get(k.upper() + '_PASSPHRASE')
+        if p == None:
+            keys[k].passphrase = sslib.interface.get_password(
+                prompt='Enter password for {0}: '.format(k), confirm=True)
+        else:
+            keys[k].passphrase = p
+
+        repo_tool.generate_and_write_ecdsa_keypair(os.path.join(
+            trustdir, name, keystore_dir, k), password=keys[k].passphrase)
```

Comment:
The example repository tool generates the keys for all repositories in the same directory (I assume for reusing the root key?)

I am still evaluating whether to:

- keep the convention from the TUF reference implementation
- mimic the behaviour of Notary and offer to create a new root key per repository.

I think it is still early in this tool to take this decision, but we should have this conversation.

Reply:
Sharing the same root key by default per repo makes sense, considering that most people are likely to have a separate repo per bundle, following the DCT model. However, we should give an option to override this.